s4e-backend/src/test/java/pl/cyfronet/s4e/controller/SearchControllerTest.java

```diff
@@ -86,8 +88,38 @@ public void shouldGetSceneBySearchEndpoint() throws Exception {
                 .andExpect(jsonPath("$.length()", is(equalTo(limit))))
                 .andExpect(jsonPath("$[0].artifacts", containsInAnyOrder(
                         "RGB_16b", "RGBs_8b", "checksum", "manifest", "metadata", "quicklook", "product_archive"
-                )))
-        ;
+                )));
+    }
+
+    @Test
+    public void shouldntGetSceneByBadByLimit() throws Exception {
+        mockMvc.perform(get(API_PREFIX_V1 + "/search")
+                .param("limit", "1000"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()", is(equalTo(30))));
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void shouldReturnErrorsForParam(String param, String value, String text) throws Exception {
+        mockMvc.perform(get(API_PREFIX_V1 + "/search")
+                .param(param, value))
+                .andExpect(status().isBadRequest())
+                .andExpect(jsonPath("$." + param, is(Arrays.asList(text))));
+    }
+
+    private static Stream<Arguments> shouldReturnErrorsForParam() {
+        return Stream.of(
+                Arguments.of(LIMIT, "abc", "Limit musi być liczbą dodatnią"),
+                Arguments.of(LIMIT, "-1", "Limit musi być liczbą dodatnią"),
+                Arguments.of(OFFSET, "abc", "Offset musi być liczbą dodatnią"),
+                Arguments.of(OFFSET, "-1", "Offset musi być liczbą dodatnią"),
+                Arguments.of(CLOUD_COVER, "abc", "Cloud cover musi być liczbą z zakresu [0 - 100]"),
+                Arguments.of(CLOUD_COVER, "-1", "Cloud cover musi być liczbą z zakresu [0 - 100]"),
+                Arguments.of(CLOUD_COVER, "1000", "Cloud cover musi być liczbą z zakresu [0 - 100]"),
+                Arguments.of(INGESTION_FROM, "2019-11-08", "Zły format daty: `2019-11-08`"),
+                Arguments.of(INGESTION_FROM, "Hakuna matata", "Zły format daty: `Hakuna matata`")
```

Comment:
You've missed the correct parameters here. I'd put them in the same test.
Also, I'd test more thoroughly the edges of the abstraction classes, e.g. cloud cover = -1, 0, 100, 101, NaN, +inf, -inf.